vaadin-date-time-picker-flow/src/main/java/com/vaadin/flow/component/datetimepicker/DateTimePicker.java

```diff
@@ -538,4 +601,45 @@ public void setDatePickerI18n(DatePickerI18n i18n) {
         this.i18n = i18n;
         datePicker.setI18n(i18n);
     }
+
+    /**
+     * Sets whether the date time picker is marked as input required.
+     *
+     * @param required
+     *        the value of the required to be set
+     */
+    public void setRequired(boolean required) {
+        getElement().setProperty("required", required);
+        this.required = required;
+    }
+
+    @Override
+    public void setRequiredIndicatorVisible(boolean requiredIndicatorVisible) {
+        super.setRequiredIndicatorVisible(requiredIndicatorVisible);
+        this.required = requiredIndicatorVisible;
+    }
+
+    /**
+     * Gets whether the date time picker is marked as input required.
+     * <p>
+     * This property is not synchronized automatically from the client side, so
+     * the returned value may not be the same as in client side.
+     *
+     * @return {@code true} if the input is required, {@code false} otherwise
+     */
+    public boolean isRequired() {
+        return getElement().getProperty("required", false);
+    }
+
+    /**
+     * Gets the validity of the date time picker value.
+     *
+     * @return the current validity of the value.
+     */
+    private boolean isInvalid(LocalDateTime value) {
+        final boolean isRequiredButEmpty = required && Objects.equals(getEmptyValue(), value);
+        final boolean isGreaterThanMax  = value != null && max != null && value.isAfter(max);
+        final boolean isSmallerThenMin = value != null && min != null && value.isBefore(min);
+        return isRequiredButEmpty || isGreaterThanMax || isSmallerThenMin;
```

Comment:
Naming typo is fixed.
The same typo occurs in `vaadin-date-picker-flow`. Should be fixed there as well. As for the naming, would like to stay aligned with other components. Otherwise change all of those at once.

https://github.com/vaadin/vaadin-date-picker-flow/blob/3b52c4bb253b4ab28dfbf0639fd18a80a55136c8/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java#L408

Reply:
Thanks. Imo it's irrelevant to try to keep consistency of function scoped temporary variables between components and I'd still prefer making things better (even if just slightly) as we go when we need to work on something and notice it could be better without being held back by keeping consistency in all possible places where it doesn't matter much.

I do agree it would be good to update everywhere where this same code is duplicated.

It's ok for now.